pde/pdes/pde.py

```diff
@@ -185,11 +185,13 @@ def __init__(
         # setup boundary conditions
         if bc_ops is None:
             bcs = {"*:*": bc}
-        else:
+        elif isinstance(bcs, dict):
             bcs = dict(bc_ops)
             if "*:*" in bcs and bc != "auto_periodic_neumann":
                 self._logger.warning("Found default BCs in `bcs` and `bc_ops`")
             bcs["*:*"] = bc  # append default boundary conditions
+        else:
+            raise TypeError('`bc_ops` must be a dictionary')
```

Comment:
You can improve it by adding `"but got {type(bc_ope)}"`

Reply:
Good idea! Thanks